app/views/submissions/index.haml

```diff
@@ -27,4 +27,10 @@
             %td= link_to submission.created_at, assignment_submission_path(@assignment, submission)
             %td= submission.uploaded_by
             %td= submission.contributor_names
-            %td= submission.status
+            - unless submission.checks_completed?
+              %td
+                %i.fa.fa-spinner.fa-spin.fa-2x
+                :javascript
+                  setTimeout(function() { location.reload() }, 500)
+            - else
+              %td= submission.status
```

Comment:
I like the idea. You can add the spinner to the submission_decorator.rb. About the javascript, be careful. If there are a lot of items that aren't completed yet, there will be 1 piece of JS per item. Perhaps add this to the top of the template file?

Reply:
I have a question, could this be done using some kind of async function? What I understand is that each submission will be rerun but the time in which the rerunning finishes for each one will be different. Doesn't that mean that even if the javascript is placed at the top, it may happen that some submissions could fail to reload?